backend/mach.ml

```diff
@@ -258,7 +258,7 @@ let equal_integer_operation left right =
   | Iadd,  Iadd  -> true
   | Isub,  Isub  -> true
   | Imul,  Imul  -> true
-  | Imulh, Imulh -> true
+  | Imulh { signed = l }, Imulh  { signed = r } -> Bool.equal l r
```

Comment:
(Use `left`/`right`, like the surrounding code.)

Reply:
Fixed.